fcrepo-http-api/src/main/java/org/fcrepo/http/api/responses/StreamingBaseHtmlProvider.java

```diff
@@ -236,32 +210,17 @@ private Template getTemplate(final Model rdf, final Node subject,
                                   .findFirst();
 
         if (!template.isPresent()) {
-            LOGGER.trace("Attempting to discover mixin types of node for resource in question: {}", subject);
-            template = rdf.listObjectsOfProperty(createResource(subject.getURI()),
-                                                 createProperty(getRDFNamespaceForJcrNamespace(JCR_NAMESPACE) +
-                                                     "mixinTypes"))
-                          .toList().stream()
-                          .map(x -> x.asLiteral().getLexicalForm())
+            LOGGER.trace("Attempting to discover RDF types of node for resource in question: {}", subject);
+            template = ofNullable(getFirst(iter(rdf.listObjectsOfProperty(subjectURI, type))
```

Comment:
This seems better to me, although I honestly don't see much value to `Optional` here. A possibly `null`-valued variable actually seems shorter and cleaner. If it's not actually the return type of a method, I tend to avoid `Optional`.